### PR TITLE
DSL Script conversion

### LINKED ISSUE #563: Update disasm_script.py
Update the script to better work with the new syntax, as a few Macros are currently receiving faulty data or are missing relevant data from `enums.h`

- [x] **GotoMap:** An `EVT_CALL` that calls `GotoMap` will have a faulty pointer as the first argument, that doesn't resembel a map-string. Most likely an issue with the disassembly that it points to some function instead of the rodata containing the string.
- [x] **enums.h:** Most of the new enums that have been created in any earlier PR have not been included into the disasm process and are thus not replaced correctly.
  - [x] **StoryProgress:** Does not seem to be working when used in a Switch-Case currently. See `arn_03/header.c` and `arn_04/header.c` for an example
  - [x] **Missing elements:** Keep the value in its original format (dec or hex) if the respective enum doesn't have a fitting element, instead of turning them all into decimals.
- [X] **Symbols:** Output the symbol name when using the script directly like `EvtSource D_80242504 = {...` 

List is most definitely incomplete and will be updated as new issues arise.

### PR DESCRIPTION
**Pull request by Brotenko**
Resolves #563 
Working towards #368 and #564

Updates `update_evts` and `disasm_script` to function better and have a few QoL improvements.
Also do some conversion to the new macro syntax.

### CODE REVIEW

**Comment by BowserSlug** [comment]
⚠️ This PR introduces 10 warnings:
- src/world/area_arn/arn_02/events.c:181: warning: missing braces around initializer for `arn_02_npcGroup_802418AC.itemDrops[0]'
- src/world/area_arn/arn_03/events.c:1124: warning: initialization makes pointer from integer without a cast
- src/world/area_arn/arn_04/events.c:141: warning: missing braces around initializer for `arn_04_npcGroup_80243840.itemDrops[0]'
- src/world/area_arn/arn_04/events.c:175: warning: missing braces around initializer for `arn_04_npcGroup_80243A30.itemDrops[0]'
- src/world/area_arn/arn_04/events.c:610: warning: initialization makes pointer from integer without a cast
- src/world/area_arn/arn_04/events.c:638: warning: implicit declaration of function `arn_04_func_80241040_BE47F0'
- src/world/area_arn/arn_07/events.c:1382: warning: initialization makes pointer from integer without a cast
- src/world/area_arn/arn_07/events.c:1410: warning: implicit declaration of function `arn_07_func_80240B00_BED8F0'
- src/world/area_arn/arn_09/events.c:204: warning: implicit declaration of function `play_model_animation'
- src/world/area_arn/arn_10/events.c:220: warning: missing braces around initializer for `arn_10_npcGroup_802409C8.itemDrops[0]'
- src/world/area_arn/arn_11/events.c:67: warning: initialization makes integer from pointer without a cast
- src/world/area_arn/arn_11/events.c:67: warning: initialization makes integer from pointer without a cast
- src/world/area_arn/arn_12/events.c:14: warning: initialization makes integer from pointer without a cast
- src/world/area_arn/arn_12/events.c:14: warning: initialization makes integer from pointer without a cast
- src/world/area_arn/arn_12/events.c:220: warning: missing braces around initializer for `arn_12_npcGroup_802409CC.itemDrops[0]'
- src/world/area_arn/arn_12/events.c:29: warning: initialization makes integer from pointer without a cast
- src/world/area_arn/arn_12/events.c:29: warning: initialization makes integer from pointer without a cast
- src/world/area_arn/arn_13/events.c:14: warning: initialization makes integer from pointer without a cast
- src/world/area_arn/arn_13/events.c:14: warning: initialization makes integer from pointer without a cast
- src/world/area_arn/arn_13/events.c:221: warning: missing braces around initializer for `arn_13_npcGroup_802409DC.itemDrops[0]'
- src/world/area_arn/arn_13/events.c:29: warning: initialization makes integer from pointer without a cast
- src/world/area_arn/arn_13/events.c:29: warning: initialization makes integer from pointer without a cast

File: include/enums.h [hunk 1]
```diff
@@ -1544,37 +1544,6 @@ enum DamageTypes {
...
-    DAMAGETYPE_PEACH_BEAM                      = 0x00800000,
-    DAMAGETYPE_MULTI_BOUNCE                    = 0x01000000,
-    DAMAGETYPE_UNBLOCKABLE                     = 0x02000000,
-    DAMAGETYPE_SPIN_SMASH                      = 0x04000000,
-    DAMAGETYPE_IGNORE_DEFENSE                  = 0x08000000,
-    DAMAGETYPE_NO_CONTACT                      = 0x10000000,
-    DAMAGETYPE_NO_OTHER_DAMAGE_POPUPS          = 0x20000000,
-    DAMAGETYPE_STATUS_ALWAYS_HITS              = 0x40000000,
-    DAMAGETYPE_TRIGGER_LUCKY                   = 0x80000000,
-};
```
Comment (nanaian) [inline include/enums.h:1576]: Why remove this?

Comment (Brotenko) [reply include/enums.h:1576]: We have a duplicate one in there with the same name and everything 😬

**Comment by nanaian** [comment]
I'm good with this PR assuming my review comment above is explained and the added warnings - probably due to a lack of EVT_PTR are fixed.

File: include/enums.h (hunk 1 above)
Comment (nanaian) [reply include/enums.h:1576]: Oof, gotcha :+1:

**Comment by Brotenko** [comment]
I was able to resolve most warnings I introduced, the rest of the warnings I looked at (not all of them because there are a lot more than 8 for some reasons) seem to be issues that existed before tho? Really confusing, so unless you **absolutely** need me to resolve those, I'd leave it like this.

**Review by nanaian — APPROVED** [review]

**Review by ethteck — APPROVED** [review]
LGTM! Nice work. The only thing I'd add is that before anyone does more batch conversion work, please check warnings as you go because I fear there's a warning-causing thing in the new script that's going to end up propagating warnings all across the codebase, so we should try to catch and fix it before we end up doing a ton of these and then having to manually fix each one

**Pull request merged**